### PR TITLE
Fixes for Napatech vendor code

### PR DESCRIPTION
**Pull request by ralpheastwood**
- [x] I have read the contributing guide lines at https://docs.suricata.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Changes:
- Replaces #9339
- Generalise NUMA configuration recommendation for >4 numa nodes
- Cleanup some log messages

### CODE REVIEW

File: src/source-napatech.c [hunk 1]
```diff
@@ -125,35 +129,65 @@ SC_ATOMIC_DECLARE(uint64_t, flow_callback_tcp_pkts);
...
     SC_ATOMIC_INIT(total_tallied);
     SC_ATOMIC_INIT(stream_count);
 
-    SC_ATOMIC_INIT(numa0_count);
-    SC_ATOMIC_INIT(numa1_count);
-    SC_ATOMIC_INIT(numa2_count);
-    SC_ATOMIC_INIT(numa3_count);
+    numa_detect = SCMalloc(sizeof(*numa_detect) * (numa_max_node() + 1));
+    if (numa_detect == NULL) {
+        FatalError("Failed to allocate memory for numa detection array: %s",
```
Comment (jlucovsky) [inline src/source-napatech.c:145]: `FatalError` terminates suricata so the `return` following this call isn't neede.

File: src/util-napatech.c [hunk 2]
```diff
@@ -1454,8 +1452,14 @@ uint32_t NapatechSetupTraffic(uint32_t first_stream, uint32_t last_stream)
             if (strchr(port->val, '-')) {
                 stream_spec = CONFIG_SPECIFIER_RANGE;
 
-                ByteExtractStringUint8(&ports_spec.first[iteration], 10, 0, port->val);
-                ByteExtractStringUint8(&ports_spec.second[iteration], 10, 0, strchr(port->val, '-')+1);
+                if (ByteExtractStringUint8(&ports_spec.first[iteration], 10, 0, port->val) == -1) {
+                    FatalError("Invalid value in napatech.ports specification in conf file.");
```
Comment (jlucovsky) [inline src/util-napatech.c:1456]: Suggest including the invalid value in the output message.

File: src/source-napatech.c (hunk 1 above)
Comment (ralpheastwood) [reply src/source-napatech.c:145]: Thanks fixed.

**Review by jlucovsky — COMMENTED** [review]
Pre-existing issue -- null pointer deref possible
https://github.com/OISF/suricata/pull/9389/files#diff-91667e39c584698b3a5816374396a1223ae0132be531a29119d40e092c6258efL930-R943

File: src/source-napatech.c [hunk 3]
```diff
@@ -928,17 +930,16 @@ TmEcode NapatechPacketLoop(ThreadVars *tv, void *data, void *slot)
         }
 
         Packet *p = PacketGetFromQueueOrAlloc();
```
Comment (ralpheastwood) [inline src/source-napatech.c:932]: Fixed this by moving the statements after the null-check.

**Comment by jlucovsky** [comment]
These changes look sound and they compile/link fine for me

@ralpheastwood What kind of testing with a physical NIC have you done?

**Comment by jlucovsky** [comment]
@ralpheastwood Review the steps here to fixup the formatting for your changes.

I use the following one-liner to fixup the formatting for my PR changes (requires them to be committed)
`git rebase master -x 'git clang-format HEAD~'`

**Comment by ralpheastwood** [comment]
Hi @jlucovsky, I used `scripts/clang-format.sh rewrite-branch` - seems to have done the trick.

**Comment by ralpheastwood** [comment]
Hi @jlucovsky  - the CI is failing on the Debian 12 builder, but seems to be an unrelated issue.  I'll try and rebase with master.

**Comment by catenacyber** [comment]
CI rust check failure is unrelated cf https://github.com/OISF/suricata/pull/9416/commits/08721d59570eb7cd8a892e0c8b28b9be0bfd2ec5

**Comment by victorjulien** [comment]
Thanks @ralpheastwood, can you do a new clean and rebased PR? I think we're then ready for merge.

**Pull request closed without merging**